src/Engine/Rendering/Pipelines/OffscreenPipeline.cpp

```
@@ -443,60 +474,116 @@ void OffscreenPipeline::createDescriptorSets(VkDescriptorPool descriptorPool, ui
 
 void OffscreenPipeline::initShaderStage() {
 	// Loads shader bytecode onto buffers
-		// Vertex shader
-	m_vertShaderBytecode = FilePathUtils::ReadFile(ShaderConst::VERTEX);
-	Log::Print(Log::T_SUCCESS, __FUNCTION__, ("Loaded vertex shader! SPIR-V bytecode file size is " + std::to_string(m_vertShaderBytecode.size()) + " (bytes)."));
-	m_vertShaderModule = createShaderModule(m_vertShaderBytecode);
+	static std::function<void(VkShaderModule &, const std::string &, const std::string)> loadShader = [this](VkShaderModule &module, const std::string &shaderPath, const std::string shaderPurpose) {
```

Cppcheck (2.20.0)

[performance] Function parameter 'shaderPurpose' should be passed by const reference.

(passedByValue)